provider/azure/config_test.go

```diff
@@ -17,51 +17,29 @@ limitations under the License.
 package azure
 
 import (
-	"fmt"
-	"io/ioutil"
-	"os"
-	"reflect"
 	"testing"
 
-	"github.com/Azure/go-autorest/autorest/azure"
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/cloud"
 )
 
-func TestGetAzureEnvironmentConfig(t *testing.T) {
-	tmp, err := ioutil.TempFile("", "azureconf")
-	if err != nil {
-		t.Errorf("couldn't write temp file %v", err)
-	}
-	defer os.Remove(tmp.Name())
-
+func TestGetCloudConfiguration(t *testing.T) {
 	tests := map[string]struct {
-		cloud string
-		err   error
+		cloudName string
+		expected  cloud.Configuration
 	}{
-		"AzureChinaCloud":   {"AzureChinaCloud", nil},
-		"AzureGermanCloud":  {"AzureGermanCloud", nil},
```

Comment:
German Cloud does not exist anymore so it has been removed from the configuration.
https://github.com/Azure/azure-sdk-for-go/issues/19112